### ISSUE TITLE
target_humidity not specify but appear in thermostat in HA

### ISSUE THREAD

**Issue by devildant** [litinoveweedle/hass-template-climate#22]
### The problem

Hi, 
I hadn't paid attention but I have the option to define the target humidity, however I don't have this option on my air conditioner, how can I remove this option from the thermostat ha?

![image](https://github.com/litinoveweedle/hass-template-climate/assets/8700106/0a53723f-71e6-45b0-9919-79afa8ba08e6)

```
- platform: climate_template
  name: Climatisation Chambre Climate Template
  unique_id: climatisation_chambre_climate_template
  # unique id not working but mendatory for working
  icon_template: mdi:air-conditioner
  hvac_modes:
    - "dry"
    - "off"
    - "cool"
    - "fan_only"
  fan_modes:
    - "Auto"
    - "20%"
    - "40%"
    - "60%"
    - "80%"
    - "100%"
    - "Silencieux"
    - "Turbo"
  swing_modes:
    - "Position 1"
    - "Position 2"
    - "Position 3"
    - "Position 4"
    - "Position 5"
    - "Activé"
    - "Désactivé"
  min_temp: 16
  max_temp: 31
  availability_template: "{{ states('switch.localtuya_integration_climatisation_chambre_power') in ['on', 'off'] }}"
  current_temperature_template: "{{ states('sensor.localtuya_integration_climatisation_chambre_temperature_courante') }}"
  set_hvac_mode:
    - service: automation.trigger
      data:
        skip_condition: false
      target:
        entity_id: automation.0011_climate_template_climatisation_chambre_set_hvac_mode
  set_fan_mode:
    - service: automation.trigger
      data:
        skip_condition: false
      target:
        entity_id: automation.0011_climate_template_climatisation_chambre_set_fan_mode
  set_swing_mode:
    - service: automation.trigger
      data:
        skip_condition: false
      target:
        entity_id: automation.0011_climate_template_climatisation_chambre_set_swing_mode
  set_temperature:
    - service: automation.trigger
      data:
        skip_condition: false
      target:
        entity_id: automation.0011_climate_template_climatisation_chambre_set_temperature
```

### What version of Template Climate has the issue?

0.7.1

### What version of Home Assistant are you running?

2024.6.3

### What type of installation are you running?

Home Assistant OS

**Comment by devildant** [comment]
I think it's a small oversight on line 530 in climate.py, you activate the feature even if _template_target_humidity is not defined.

maybe fix like that : 
```
if HVACMode.DRY in self._attr_hvac_modes:
    if self._template_target_humidity:
        self._attr_supported_features |= ClimateEntityFeature.TARGET_HUMIDITY
```

**Comment by devildant** [comment]
![image](https://github.com/litinoveweedle/hass-template-climate/assets/8700106/1652686f-31e1-4d01-b04c-7a40134c2835)

I tested successfully but I'm not sure, with your presets_features option (I don't use it)

**Comment by litinoveweedle** [comment]
Hello,
it is actually not that simple. There are two ways ho to set target humidity:

- via template
- via HA GUI

So the fact, that humidity template is not defined doesn't mean, that someone is not using HA to set humidity. So far the logic was: if there is DRY mode in hvac mode then we suppose, that target humidity could be set, While I understand, that some devices might have DRY mode, but not allow to set target humidity (for example some AC units), then what you propose would be more limiting to users, who need this functionality and will not be able to use it, if they do not use template to set target humidity (which is not always the case).

So proposed way would be to implement whole condition:

```
            if  (
                self._action_humidity
                or self._template_target_humidity
                or self._presets_features & ClimateEntityPresetFeature.TARGET_HUMIDITY
            ):
                self._attr_supported_features |= ClimateEntityFeature.TARGET_HUMIDITY
            else:
                _LOGGER.info(
```

The same logic should be implemented for TARGET_TEMPERATURE as well. I will evaluate this and if it would be OK I will implement this in the next release.

**Comment by devildant** [comment]
> Hello, it is actually not that simple. There are two ways ho to set target humidity:
> 
> * via template
> * via HA GUI
> 
> So the fact, that humidity template is not defined doesn't mean, that someone is not using HA to set humidity. So far the logic was: if there is DRY mode in hvac mode then we suppose, that target humidity could be set, While I understand, that some devices might have DRY mode, but not allow to set target humidity (for example some AC units), then what you propose would be more limiting to users, who need this functionality and will not be able to use it, if they do not use template to set target humidity (which is not always the case).
> 
> So proposed way would be to implement whole condition:
> 
> ```
>             if  (
>                 self._action_humidity
>                 or self._template_target_humidity
>                 or self._presets_features & ClimateEntityPresetFeature.TARGET_HUMIDITY
>             ):
>                 self._attr_supported_features |= ClimateEntityFeature.TARGET_HUMIDITY
>             else:
>                 _LOGGER.info(
> ```
> 
> The same logic should be implemented for TARGET_TEMPERATURE as well. I will evaluate this and if it would be OK I will implement this in the next release.

your condition seems to me more complete.
it was at least to give a quick forward, and help a little with the resolution.

I suspected my solution was a bit lightweight.
that's why I didn't do a "PR"
+1 for apply same logcal for the temperature

**Comment by litinoveweedle** [comment]
hello, fixed in [this](https://github.com/litinoveweedle/hass-template-climate/tree/targets_conditional_support) branch, please copy [climate.py](https://github.com/litinoveweedle/hass-template-climate/blob/targets_conditional_support/custom_components/climate_template/climate.py), restart HA test and report back please. Thank you.

**Comment by devildant** [comment]
> hello, fixed in [this](https://github.com/litinoveweedle/hass-template-climate/tree/targets_conditional_support) branch, please copy [climate.py](https://github.com/litinoveweedle/hass-template-climate/blob/targets_conditional_support/custom_components/climate_template/climate.py), restart HA test and report back please. Thank you.

![image](https://github.com/litinoveweedle/hass-template-climate/assets/8700106/9a08679b-9065-478f-9d18-21021c2cff1c)

👍

**Comment by litinoveweedle** [comment]
Released in [0.7.2](https://github.com/litinoveweedle/hass-template-climate/releases/tag/0.7.2)
